qiskit_experiments/calibration_management/calibrations.py

```diff
@@ -1035,6 +1023,49 @@ def get_parameter_value(
         # 5) Return the most recent parameter.
         return max(enumerate(candidates), key=lambda x: (x[1].date_time, x[0]))[1].value
 
+    def _standardize_assign_params(
+        self,
+        assign_params: Dict,
+        qubits: Tuple[int, ...],
+        schedule_name: str,
+    ) -> Dict[ParameterKey, ParameterValueType]:
+        """Standardize the format of manually specified parameter assignments.
+
+        Users specify parameter assignment dictionaries as tuples. This function (a) converts
+        these tuples to ``ParameterKey`` instances and (b) looks up parameter links between
+        schedules and adds them to the standardized returned parameter assignment dictionary.
+
+        Args:
+            assign_params: The dictionary that specifies how to assign parameters.
+            qubits: The qubits for which to get a schedule.
+            schedule_name: The name of the schedule in which the parameters can be found.
+
+        Returns:
+            A dict with :class:`.ParameterKey`s as keys and :class:`.ParameterValueType`s as values.
+        """
+        assign_params_, linked_assign_params = dict(), dict()
+
+        if assign_params:
+            for param, value in assign_params.items():
+                if isinstance(param, str):
+                    assign_params_[ParameterKey(param, qubits, schedule_name)] = value
+                else:
+                    assign_params_[ParameterKey(*param)] = value
+
+            # Add parameter links for automatic linking.
+            for key, value in assign_params_.items():
```

Comment:
This is to explicitly show duplicated parameters with different names (keys) based off of the uuid equivalency?

Reply:
This makes sure that schedules with references keep parameter links when we specify a parameter assignment dictionary. For example, consider an ECR built from references in which the amplitude of `cr45p` is `amp` (where `amp` is a parameter) and the amplitude of `cr45m` is `-amp`. This is the case of the present ECR lib. I want to specify the amplitude in an assignment dictionary using only the `cr45p` pulse and have the `cr45m` assigned accordingly. This code takes care of that. Take a look at the tests in `test_calibrations.TestCoupledAssigning` to see the various use cases.